source/common/upstream/upstream_impl.cc

```diff
@@ -119,12 +119,23 @@ parseExtensionProtocolOptions(const envoy::api::v2::Cluster& config) {
   for (const auto& iter : config.extension_protocol_options()) {
     const std::string& name = iter.first;
     const ProtobufWkt::Struct& config_struct = iter.second;
+    Server::Configuration::ProtocolOptionsConsumer* factory = nullptr;
 
-    auto& factory = Envoy::Config::Utility::getAndCheckFactory<
-        Server::Configuration::NamedNetworkFilterConfigFactory>(name);
+    factory = Registry::FactoryRegistry<
+        Server::Configuration::NamedNetworkFilterConfigFactory>::getFactory(name);
+    if (factory == nullptr) {
+      factory = Registry::FactoryRegistry<
+          Server::Configuration::NamedHttpFilterConfigFactory>::getFactory(name);
+    }
+
+    if (factory == nullptr) {
+      throw EnvoyException(fmt::format(
+          "Didn't find a registered network or http implementation for name: '{}'", name));
```

Comment:
Let's add "filter" between http and implementation.

Reply:
sounds good